subsys/bluetooth/host/classic/avrcp.c

```diff
@@ -0,0 +1,365 @@
+				BT_SDP_TYPE_SIZE(BT_SDP_UUID16),
+				BT_SDP_ARRAY_16(BT_SDP_AV_REMOTE_SVCLASS)
+			},
+			{
+				BT_SDP_TYPE_SIZE(BT_SDP_UINT16),
+				BT_SDP_ARRAY_16(0x0106U)
+			},
+			)
+		},
+		)
+	),
+	BT_SDP_SUPPORTED_FEATURES(0x0003U), /* Category 1 & 2 */
+	/* O: Provider Name not presented */
+	BT_SDP_SERVICE_NAME("AVRCP Target"),
+};
+
+static struct bt_sdp_record avrcp_tg_rec = BT_SDP_RECORD(avrcp_tg_attrs);
+#endif /* CONFIG_BT_AVRCP_TARGET */
+
+#if defined(CONFIG_BT_AVRCP_CONTROLLER)
+static struct bt_sdp_attribute avrcp_ct_attrs[] = {
+	BT_SDP_NEW_SERVICE,
+	BT_SDP_LIST(
+		BT_SDP_ATTR_SVCLASS_ID_LIST,
+		BT_SDP_TYPE_SIZE_VAR(BT_SDP_SEQ8, 6),
+		BT_SDP_DATA_ELEM_LIST(
+		{
+			BT_SDP_TYPE_SIZE(BT_SDP_UUID16),
+			BT_SDP_ARRAY_16(BT_SDP_AV_REMOTE_SVCLASS)
+		},
+		{
+			BT_SDP_TYPE_SIZE(BT_SDP_UUID16),
+			BT_SDP_ARRAY_16(BT_SDP_AV_REMOTE_CONTROLLER_SVCLASS)
+		},
+		)
+	),
+	BT_SDP_LIST(
+		BT_SDP_ATTR_PROTO_DESC_LIST,
+		BT_SDP_TYPE_SIZE_VAR(BT_SDP_SEQ8, 16),
+		BT_SDP_DATA_ELEM_LIST(
+		{
+			BT_SDP_TYPE_SIZE_VAR(BT_SDP_SEQ8, 6),
+			BT_SDP_DATA_ELEM_LIST(
+			{
+				BT_SDP_TYPE_SIZE(BT_SDP_UUID16),
+				BT_SDP_ARRAY_16(BT_SDP_PROTO_L2CAP)
+			},
+			{
+				BT_SDP_TYPE_SIZE(BT_SDP_UINT16),
+				BT_SDP_ARRAY_16(BT_UUID_AVCTP_VAL)
+			},
+			)
+		},
+		{
+			BT_SDP_TYPE_SIZE_VAR(BT_SDP_SEQ8, 6),
+			BT_SDP_DATA_ELEM_LIST(
+			{
+				BT_SDP_TYPE_SIZE(BT_SDP_UUID16),
+				BT_SDP_ARRAY_16(BT_UUID_AVCTP_VAL)
+			},
+			{
+				BT_SDP_TYPE_SIZE(BT_SDP_UINT16),
+				BT_SDP_ARRAY_16(0X0104u)
+			},
+			)
+		},
+		)
+	),
+	/* C1: Browsing not supported */
+	BT_SDP_LIST(
+		BT_SDP_ATTR_PROFILE_DESC_LIST,
+		BT_SDP_TYPE_SIZE_VAR(BT_SDP_SEQ8, 8),
+		BT_SDP_DATA_ELEM_LIST(
+		{
+			BT_SDP_TYPE_SIZE_VAR(BT_SDP_SEQ8, 6),
+			BT_SDP_DATA_ELEM_LIST(
+			{
+				BT_SDP_TYPE_SIZE(BT_SDP_UUID16),
+				BT_SDP_ARRAY_16(BT_SDP_AV_REMOTE_SVCLASS)
+			},
+			{
+				BT_SDP_TYPE_SIZE(BT_SDP_UINT16),
+				BT_SDP_ARRAY_16(0x0106U)
+			},
+			)
+		},
+		)
+	),
+	BT_SDP_SUPPORTED_FEATURES(0x0003U), /* Category 1 & 2 */
+	/* O: Provider Name not presented */
+	BT_SDP_SERVICE_NAME("AVRCP Controller"),
+};
+
+static struct bt_sdp_record avrcp_ct_rec = BT_SDP_RECORD(avrcp_ct_attrs);
+#endif /* CONFIG_BT_AVRCP_CONTROLLER */
+
+static struct bt_avrcp *get_new_connection(struct bt_conn *conn)
+{
+	uint8_t i;
+
+	if (!conn) {
+		LOG_ERR("Invalid Input (err: %d)", -EINVAL);
+		return NULL;
+	}
+
+	for (i = 0; i < ARRAY_SIZE(avrcp_connection); i++) {
+		if (!avrcp_connection[i].session.br_chan.chan.conn) {
+			memset(&avrcp_connection[i], 0, sizeof(avrcp_connection[i]));
+			return &avrcp_connection[i];
+		}
+	}
+
+	LOG_ERR("More connection cannot be supported");
+	return NULL;
+}
+
+static struct bt_avrcp *avrcp_lookup_by_conn(struct bt_conn *conn)
+{
+	uint8_t i;
+
+	if (!conn) {
+		return NULL;
+	}
+
+	for (i = 0; i < ARRAY_SIZE(avrcp_connection); i++) {
+		if (avrcp_connection[i].session.br_chan.chan.conn == conn) {
+			return &avrcp_connection[i];
+		}
+	}
+
+	return NULL;
+}
```

Comment:
You should be able to make all of this much simpler by using the `bt_conn_index()` API. That returns an array index that can be used whenever you have an array with `CONFIG_BT_MAX_CONN` elements. You get the AVRCP context then with:
```c
&avrcp_connection[bt_conn_index(conn)]
```

Reply:
> You should be able to make all of this much simpler by using the `bt_conn_index()` API. That returns an array index that can be used whenever you have an array with `CONFIG_BT_MAX_CONN` elements. You get the AVRCP context then with:
> 
> ```c
> &avrcp_connection[bt_conn_index(conn)]
> ```

Thanks, this really makes the code clean.